packages/core/src/http/requestBuilder.ts

```diff
@@ -385,14 +393,18 @@ export class DefaultRequestBuilder<BaseUrlParamType, AuthParams>
   ): void {
     this.intercept(async (req, opt, next) => interceptor(await next(req, opt)));
   }
-  public defaultToError(apiErrorCtor: ApiErrorConstructor): void {
-    this._apiErrorFactory = apiErrorCtor;
+  public defaultToError(
+    apiErrorCtor: ApiErrorConstructor,
+    message?: string
+  ): void {
+    this._apiErrorFactory = { apiErrorCtor, message };
   }
   public validateResponse(validate: boolean): void {
     this._validateResponse = validate;
   }
   public throwOn<ErrorCtorArgs extends any[]>(
     statusCode: number | [number, number],
+    isTemplate: boolean = false,
```

Comment:
Adding a new parameter here makes it a breaking change

Reply:
added another signature function using function overloading to fix this. now it wont be a breaking change for customers.